Mathlib/GroupTheory/MaximalSubgroups.lean

```diff
@@ -0,0 +1,67 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir
+-/
+
+import Mathlib.Algebra.Group.Subgroup.Basic
+import Mathlib.Order.Atoms
+
+/-! # Maximal subgroups
+
+A subgroup `IsMaximal` if it is maximal in the collection of
+proper subgroups.
+
+We prove a few basic results which are essentially copied from
+those about maximal ideals.
+
+Besides them, we have :
+* `isMaximal_iff` proves that a subgroup K of G is maximal
+if and only  if it is not ⊤ and if for all g ∈ G such that g ∉ K,
+every subgroup containing K and g is ⊤.
+
+## TODO
+
+Is it useful to write it in a greater generality?
+Will we need to write this for all `sub`-structures ?
+
+-/
+
+
+variable {G : Type*} [Group G]
+
+namespace Subgroup
+
+/-- A subgroup is maximal if it is maximal in the collection of proper subgroups. -/
+class IsMaximal (K : Subgroup G) : Prop where
```

Comment:
I see that you define it as a typeclass, then use as `(h : K.IsMaximal)`, not `[K.IsMaximal]`. If you're going to use it as a predicate, why not use `IsCoatom` instead?

Reply:
For the typeclass/predicate, it's not clear to me whether we want a class or a structure. I believe I copied what was done for maximal ideals, hence the class…